config/config.go

```diff
@@ -132,48 +117,58 @@ func (c *Config) ReadConfig(configFilePath string, flagSet *pflag.FlagSet) error
 	return nil
 }
 
-func validateConfig(config *Config) error {
+func validateConfig(config *Config, categories ...Category) error {
 	var result *multierror.Error
 
-	if config.Ethereum.URL == "" {
-		result = multierror.Append(result, fmt.Errorf(
-			"missing value for ethereum.url; see ethereum section in configuration",
-		))
-	}
-
-	if config.Ethereum.Account.KeyFile == "" {
-		result = multierror.Append(result, fmt.Errorf(
-			"missing value for ethereum.keyFile; see ethereum section in configuration",
-		))
-	}
-
-	if config.LibP2P.Port == 0 {
-		result = multierror.Append(result, fmt.Errorf(
-			"missing value for network.port; see network section in configuration",
-		))
-	}
+	for _, category := range categories {
+		switch category {
+		case Ethereum:
+			if config.Ethereum.URL == "" {
+				result = multierror.Append(result, fmt.Errorf(
+					"missing value for ethereum.url; see ethereum section in configuration",
+				))
+			}
 
-	if config.Storage.DataDir == "" {
-		result = multierror.Append(result, fmt.Errorf(
-			"missing value for storage.dataDir; see storage section in configuration",
-		))
+			if config.Ethereum.Account.KeyFile == "" {
+				result = multierror.Append(result, fmt.Errorf(
+					"missing value for ethereum.keyFile; see ethereum section in configuration",
+				))
+			}
+		case Network:
+			if config.LibP2P.Port == 0 {
+				result = multierror.Append(result, fmt.Errorf(
+					"missing value for network.port; see network section in configuration",
+				))
+			}
+		case Storage:
+			if config.Storage.DataDir == "" {
+				result = multierror.Append(result, fmt.Errorf(
+					"missing value for storage.dataDir; see storage section in configuration",
+				))
+			}
+		}
 	}
 
 	return result.ErrorOrNil()
 }
 
-// ReadEthereumConfig reads in the configuration file at `filePath` and returns
-// its contained Ethereum config, or an error if something fails while reading
-// the file.
+// ReadEthereumConfig reads in the configuration from a file specified by `--config`
+// flag and other flags provided in the `flagSet` and returns its contained Ethereum
+// config, or an error if something fails.
 //
 // This is the same as invoking ReadConfig and reading the Ethereum property
 // from the returned config, but is available for external functions that expect
 // to interact solely with Ethereum and are therefore independent of the rest of
 // the config structure.
-func ReadEthereumConfig(filePath string) (ethereum.Config, error) {
-	config := &Config{}
-	err := config.ReadConfig(filePath, nil)
+func ReadEthereumConfig(flagSet *pflag.FlagSet) (ethereum.Config, error) {
+	config := Config{}
+
+	configPath, err := flagSet.GetString("config")
 	if err != nil {
+		return ethereum.Config{}, fmt.Errorf("failed to read config file path from command flag: %w", err)
```

Comment:
This line is over 100 characters long. Could we split it somehow, e.g.:
```
		return ethereum.Config{}, fmt.Errorf(
			"failed to read config file path from command flag: %w",
			err,
		)
```

Reply:
f5a22a044505451e2713ea75939340d45b12a938